integration-test/2047-restore-early-lake-labels.py

```diff
@@ -0,0 +1,117 @@
+            '37.80395991667648))'),
+            {
+            u'way_area': u'6936.45',
+            u'natural': u'water',
+            u'source': u'openstreetmap.org',
+            u'water': u'reservoir',
+            u'name': u'Francisco Reservoir',
+            u'covered': u'no'}))
+
+        self.assert_has_feature(
+            16, 10481, 25324, 'water',
+            {
+                'kind': 'water',
+                'reservoir': True,
+                'kind_detail': 'lake'
+            })
+
+    def test_water_lagoon(self):
+        self.generate_fixtures(dsl.way(1, wkt_loads(
+            'POLYGON ((-122.421611012467 37.80395991667648, '
+            '-122.420227786592 37.80413210674329, -122.420159873957 '
+            '37.80408277780808, -122.420117832801 37.80383748120349, '
+            '-122.42154480663 37.80365748294528, -122.421611012467 '
+            '37.80395991667648))'),
+            {
+            u'natural': u'water',
+            u'way_area': u'6936.45',
+            u'source': u'openstreetmap.org',
+            u'water': u'lagoon',
+            u'name': u'Francisco Reservoir',
+            u'covered': u'no'}))
+
+        self.assert_has_feature(
+            16, 10481, 25324, 'water',
+            {
+                'kind': 'water',
+                'alkaline': True,
+                'kind_detail': 'lake'
+            })
+
+    def test_water_ditch(self):
+        self.generate_fixtures(dsl.way(1, wkt_loads(
+            'POLYGON ((-122.421611012467 37.80395991667648, -122.420227786592 37.80413210674329, -122.420159873957 37.80408277780808, -122.420117832801 37.80383748120349, -122.42154480663 37.80365748294528, -122.421611012467 37.80395991667648))'),
+            {
+            u'natural': u'water',
+            u'way_area': u'6936.45',
+            u'source': u'openstreetmap.org',
+            u'water': u'ditch',
+            u'name': u'Francisco Reservoir',
+            u'covered': u'no'}))
+
+        self.assert_has_feature(
+            16, 10481, 25324, 'water',
+            {
+                'kind': 'water',
+                'kind_detail': 'ditch'
+            })
+
+    def test_name_not_drop(self):
+        import dsl
+
+        for z in range(6, 13):
+            x, y = (3, 4)
+            area = 265548000
+
+            self.generate_fixtures(
+                dsl.way(1, dsl.box_area(z, x, y, area, include_boundary=True), {
+                    'name': u'Clear Lake',
+                    'natural': u'water',
+                    'type': u'multipolygon',
+                    'water': u'lake',
+                    'wikidata': u'Q1099503',
+                    'source': u'openstreetmap.org',
+                }),
+            )
+
+            self.assert_has_feature(
+                z, x, y, 'water', {
+                    'kind': 'water',
+                    'kind_detail': 'lake',
+                    'name': 'Clear Lake',
+                    'min_zoom': 6.0,
```

Comment:
For the generated label point geometry (the cruz of this PR), you want to also test for:

>                     'label_placement': True,

Otherwise you're testing the polygon which has it's own min_zoom range.

To clarify... the polygon should be visible from zoom 6 conceptually based on it's area grading of `265548000` sq meters in https://github.com/tilezen/vector-datasource/blob/master/yaml/water.yaml#L23-L46.

```
        - [ 5,   500000000 ]
        - [ 6,   200000000 ]
        - [ 7,    50000000 ]
```

For the label we do a remapping of the min_zoom based on area grading of `265548000` sq metershere: https://github.com/tilezen/vector-datasource/blob/master/queries.yaml#L843-L878, so I expect min_zoom 9 for the label.

```
           (8,     500000000),
            (9,     200000000),
            (10,     40000000),
```

Since we don't transition from NE to OSM varies based on layer, but zoom 4 presumably for water from the osm polygons table:

https://github.com/tilezen/vector-datasource/blob/master/queries.yaml#L23-L26

So to recap: 
- the lake polygon shouldn't be visible until zoom 6, and it won't have a name property ever
- the lake label shouldn't be visible until zoom 9 (but wasn't visible until zoom 12 in error before), so we are testing for it's presence at zooms 9, 10, and 11 (and that it's not present at 8, and still present at zoom 12).

Reply:
@nvkelso please see the latest commit about that integration test, the behavior is not what you described here. 

what the integ test told us are:

1. We are seeing `name` property appear since zoom 6 for the lake polygon
2. We are seeing label_placement since zoom 6

Reply:
there is a bug in the yaml file processing in the update_min_zoom function, just fixed that and then fixed the integration tests